### ISSUE TITLE
Focus should not reach <body>

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#2327]
Testing #2145

`setInterval(function() { console.log(document.activeElement); }, 2000);`
- put last editor in Tab moves focus mode via `Ctrl+M`
- tab in last editor:

![image](https://cloud.githubusercontent.com/assets/5047891/12581676/76244de8-c438-11e5-858f-d39b2acef3f5.png)

**Comment by bpasero** [comment]
@alexandrudima yeah, I am not sure I can disable this actually. maybe I can redirect the focus to the first control somewhere.

**Comment by bpasero** [comment]
@alexandrudima tried a while now to see if I can prevent keyboard to navigate into the <body> element and failed to find anything useful. The only super hack I can come up with is to listen globally to Tab-KeyUp and find out if the active element is the body and then move focus to some element within.

If anyone has ideas, they are most welcome.

**Comment by alexandrudima** [comment]
@bpasero 'focus' listener on `<body>` ?

**Comment by alexandrudima** [comment]
or there was another one 'focusin' - I forgot the exact difference. You can check if the `e.target === body`

**Comment by bpasero** [comment]
@alexandrudima best I can come up with is this:

```
window.document.body.addEventListener('blur', function(e) {
   setTimeout(function() { console.log(document.activeElement);});
}, true);
```

I do not get any focus event on the body element, so listen on blur in the capture phase and use timeout because without, activeElement is === body. Still a bit ugly.

**Comment by bpasero** [comment]
I am not really able to do this nicely, moving to backlog.

**Comment by bpasero** [comment]
Maybe this will help in Chrome 50: https://developers.google.com/web/updates/2016/03/focus-start-point

**Comment by cleidigh** [comment]
@bpasero 
This looks like it also relates to what I'm trying to do with some list  navigation improvement with
PR #24628 in dealing with tabindex issues
you still have help-wanted tag shall I take a look at it or is somebody on this?

**Comment by bpasero** [comment]
@cleidigh the issue is that when you tab long enough through the UI you end up in the body element at some point without any visual indication. If possible I wanted to disallow jumping to this element because it does not really help for navigation.

**Comment by bpasero** [comment]
Closing as I am seeing no way of doing this currently, if someone wants to fix this, PR me.

**Comment by bpasero** [comment]
This is now actually fixed with Electron 3 👍
